dotnet/src/Microsoft.AutoGen/RuntimeGateway.Grpc/Services/Grpc/GrpcGateway.cs

```diff
@@ -131,10 +155,48 @@ public async ValueTask<RegisterAgentTypeResponse> RegisterAgentTypeAsync(Registe
                 {
                     throw new RpcException(new Status(StatusCode.InvalidArgument, $"Grpc Worker Connection not found for ClientId {clientId}. Retry after you call OpenChannel() first."));
                 }
-                connection.AddSupportedType(request.Type);
-                _supportedAgentTypes.GetOrAdd(request.Type, _ => []).Add(connection);
 
-                await _gatewayRegistry.RegisterAgentTypeAsync(request, clientId, _reference).ConfigureAwait(true);
+                // If we are not supporting multiplexing, we need to ensure that the agent type is not already registered by
```

Comment:
There could be other gateways.  - the logic here only works fr this one - Shouldn't this be in the RegistryGrain.RegisterAgentType instead, and there you should use RegistryGrain.GetOrAddWorker() to find out if you have a connection already registered for this type?